bpython/curtsiesfrontend/manual_readline.py

```diff
@@ -231,6 +231,25 @@ def delete(cursor_offset, line):
     return (cursor_offset, line[:cursor_offset] + line[cursor_offset + 1 :])
 
 
+def cursor_on_closing_char_pair(cursor_offset, line):
+    """Checks if cursor sits on closing character of a pair
+    and whether its pair character is directly behind it
+    """
+    closing_char_map = {"(": ")", "{": "}", "[": "]", "'": "'", '"': '"'}
+    on_closing_char, pair_close = False, False
+    if line is None:
+        return on_closing_char, pair_close
+    if cursor_offset < len(line):
+        cur_char = line[cursor_offset]
+        prev_char = line[cursor_offset - 1]
```

Comment:
What's with the case `cursor_offset = 0`?

Reply:
i will fix this